File: iotonrails/app/models/telemetrystream.rb

class TelemetryStream < ApplicationRecord
  enum Qos: [:AtMostOnce, :AtLeastOnce, :ExactlyOnce]


  has_many :Device, class_name: 'IoTDevice'
  has_many :Sensor, class_name: 'SensorInstance'
  has_many :Schema, class_name: 'TelemetrySchema'
  has_many :MessagingEndpoint, class_name: 'MessagingEndpoint'
  has_many :RetentionPolicy, class_name: 'DataRetentionPolicy'

end

File: iotonaspdotnet/Domain/TelemetryStream.cs
using iotonaspdotnet.Domain.Contracts;

namespace iotonaspdotnet.Domain;

public class TelemetryStream
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long TelemetrystreamId { get; set; }
 public virtual string StreamName { get; set; }
 public virtual int RetentionDays { get; set; }
public virtual IoTDevice Device { get; set; }
public virtual SensorInstance Sensor { get; set; }
public virtual TelemetrySchema Schema { get; set; }
public virtual MessagingEndpoint MessagingEndpoint { get; set; }
public virtual DataRetentionPolicy RetentionPolicy { get; set; }
 public virtual MessageQoS Qos { get; set; }

    public static TelemetryStream FromRequest(TelemetryStreamRequest request) {
        return new TelemetryStream {
            Id = model.Id,
            StreamName = request.StreamName,
            RetentionDays = request.RetentionDays,
            Qos = request.Qos,
        };
}
